modules/test/playwright/tests/style-book-web/previewStyleBookOnDisplayPageTemplates.spec.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Page, expect, mergeTests} from '@playwright/test';
+
+import {isolatedSiteTest} from '../../fixtures/isolatedSiteTest';
+import {loginTest} from '../../fixtures/loginTest';
+import {pageEditorPagesTest} from '../../fixtures/pageEditorPagesTest';
+import {styleBookPageTest} from '../../fixtures/styleBookPageTest';
+import {workflowPagesTest} from '../../fixtures/workflowPagesTest';
+import {PageEditorPage} from '../../pages/layout-content-page-editor-web/PageEditorPage';
+import {DisplayPageTemplatesPage} from '../../pages/layout-page-template-admin-web/DisplayPageTemplatesPage';
+import {StyleBooksPage} from '../../pages/style-book-web/StyleBooksPage';
+import getRandomString from '../../utils/getRandomString';
+import {blogsPagesTest} from '../blogs-web/fixtures/blogsPagesTest';
+
+const test = mergeTests(
+	blogsPagesTest,
+	isolatedSiteTest,
+	loginTest(),
+	pageEditorPagesTest,
+	styleBookPageTest,
+	workflowPagesTest
+);
+
+const changeStyleAndCheckPreview = async (
+	colorHEX: string,
+	colorRGB: string,
+	displayPageTemplatesPage: DisplayPageTemplatesPage,
+	page: Page,
+	pageEditorPage: PageEditorPage,
+	site: Site,
+	styleBooksPage: StyleBooksPage,
+	displayPageTemplateName: string
+) => {
+	await styleBooksPage.goto(site.friendlyUrlPath);
+	await styleBooksPage.editStyleBook('Test Style Book Name');
+
+	await page.getByRole('button', {name: 'Color System'}).click();
+	await page.getByRole('menuitem', {name: 'General'}).click();
+
+	await styleBooksPage.updateTokenInputColor('Body Color', colorHEX);
+
+	expect(page.getByText('Saved')).toBeVisible();
+
+	await styleBooksPage.publishStyleBook();
+
+	await displayPageTemplatesPage.goto(site.friendlyUrlPath);
+
+	await displayPageTemplatesPage.editTemplate(displayPageTemplateName);
+	await page.getByLabel('Page Design Options').click();
+	await page.getByRole('tab', {name: 'Style Book'}).click();
+	await page.getByLabel('Test Style Book Name').click();
+
+	const headingId = await pageEditorPage.getFragmentId('Heading');
+
+	const fragmentColor = await pageEditorPage.getFragmentStyle({
+		fragmentId: headingId,
+		isTopperStyle: true,
+		style: 'color',
+	});
+
+	expect(fragmentColor).toBe(colorRGB);
+};
+
+test('Add a heading fragment and view the display page templates is shown in preview type selector', async ({
+	displayPageTemplatesPage,
+	page,
+	pageEditorPage,
+	site,
+	styleBooksPage,
+}) => {
+	const displayPageTemplateName = getRandomString();
+
+	await test.step('Create style book and view the fragments is selected in the preview type selector', async () => {
+		await styleBooksPage.goto(site.friendlyUrlPath);
+		await styleBooksPage.createStyleBook('Test Style Book Name');
```

Comment:
What is the criteria be used for when to add line breaks and when not to add them?

I like the criteria Brian uses:
- If the order of the lines doesn't matter and they can be sorted alphabetically, then keep them together. (IE: no line break)
- If order matters and you can't alphabetically sort the lines then add a line break between each line.

I'm open to using different criteria, but it needs to be clear and easy to understand and apply.

cc: @thiago-buarqque

Reply:
For Playwright I already saw both usages in other files, but I follow the one you just mentioned.

Reply:
@thiago-buarqque what would the other criteria for line breaks in other files be?

Reply:
Thanks @ethib137, I sorted it according to Brian's criteria.